### PR TITLE
Add browser-based receipt verifier at obsigna.dev/verify

### PR DESCRIPTION
**Pull request by ojongerius**
## What

Adds a fully client-side receipt-chain verifier at `obsigna.dev/verify` that runs entirely in the browser via WebAssembly. The verifier wraps the same Go core (`obsigna.dev/sdk/go/receipt`) used by the `obsigna receipt verify` CLI, ensuring cryptographic consistency between the browser and command-line implementations.

## Why

Users need a way to verify receipt chains without installing CLI tools or uploading sensitive data to a server. By compiling the existing Go verify core to WebAssembly, we provide:

1. **Single implementation guarantee**: No separate JavaScript reimplementation of cryptographic steps (JCS canonicalization, Ed25519 signature verification, hash-chain walking). The browser runs identical code to the CLI.
2. **Conformance gate**: A new integration test (`verify_wasm_gate_test.go`) compiles the verifier to both `wasip1` and `js` targets and asserts byte-identical output on every conformance vector, preventing drift between browser and CLI.
3. **Privacy**: All verification happens client-side; nothing is uploaded or stored.
4. **Two-ring verdict model**: Cryptographic consistency and external-anchor trust are reported separately, with unanchored chains marked as "qualified" rather than "full" (anchor verification is a tracked follow-up).

## Implementation

- **`cross-sdk-tests/verifier/verifier.go`**: Core verifier package that delegates all cryptographic work to `obsigna.dev/sdk/go/receipt`. Handles single-receipt and chain verification, parses JSON/JSONL input, and returns structured verdicts.
- **`cross-sdk-tests/cmd/verify-wasm/main.go`**: Browser entry point (GOOS=js GOARCH=wasm) that exposes `obsignaVerify(requestJSON)` as a global function.
- **`cross-sdk-tests/cmd/verify-wasm-cli/main.go`**: WASI build for conformance testing; reads request JSON from stdin, writes result to stdout.
- **`cross-sdk-tests/verify_wasm_gate_test.go`**: Integration test that compiles the verifier to wasip1, executes it under wazero, and asserts output matches native `verifier.Run` calls on all conformance vectors (valid, malformed, and edge-case corpora).
- **`site-obsigna/public/verify/`**: UI layer (HTML, CSS, JavaScript) with no verification logic—only input marshaling and result rendering.
- **`scripts/build-verify-wasm.sh`**: Build script that compiles the GOOS=js GOARCH=wasm binary and stages it with `wasm_exec.js`.

## Checklist

- [x] Tests pass for all changed components (conformance gate covers all vectors)
- [x] No real keys or secrets in the diff (only test fixtures from cross-sdk-tests)
- [x] Cross-language tests pass (conformance gate verifies WASM ↔ native identity)
- [x] AGENTS.md not updated (no project structure changes)

## Security

- [x] This PR touches crypto (verification, not signing)
- [x] Primitives and parameters reviewed: Ed25519 signature verification and JCS canonicalization delegated entirely to `obsigna.dev/sdk/go/receipt`; no new crypto implemented
- [x] All inputs validated at trust boundaries: public key parsing, receipt JSON parsing, mode inference
- [x] Edge cases tested: missing/garbage keys (ERROR), empty input (ERROR), valid receipt with wrong key (FAIL), malformed receipts (FAIL)

### CODE REVIEW

File: .github/workflows/web-verifier.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    defaults:
+      run:
+        working-directory: web-verifier
+    steps:
+      - uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+
+      - uses: actions/setup-go@4a3601121dd01d1626a1e23e37211e3254c1c06c # v6
+        with:
+          go-version: "1.26"
+
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply .github/workflows/web-verifier.yml:32]: Good catch — fixed in 702eaa9. Switched to `go-version-file: web-verifier/go.mod` so the toolchain is pinned to the exact patch recorded in the module (1.26.1). You're right that this workflow builds the *deployed* WASM artifact, so it belongs to the repo's reproducible-build convention (ADR-0035, as in collector/daemon/hook/mcp-proxy) rather than the floating `go-version: "1.26"` used by the pure-test workflows.

File: .github/workflows/site-obsigna.yml [hunk 2]
```diff
@@ -33,6 +36,17 @@ jobs:
...
         uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
 
+      # The /verify page ships a WebAssembly verifier compiled from the Go verify
+      # core. It is built fresh from source here (never committed) so the
+      # deployed binary always matches the verifier the CLI<->WASM gate checks.
+      - name: Setup Go (for the /verify WASM verifier)
+        uses: actions/setup-go@4a3601121dd01d1626a1e23e37211e3254c1c06c # v6
+        with:
+          go-version: "1.26"
+
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply .github/workflows/site-obsigna.yml:48]: Fixed in 702eaa9 — same change here: the "Setup Go" step now uses `go-version-file: web-verifier/go.mod` (the module whose `cmd/wasm` this step compiles). Since this step produces the binary that ships to obsigna.dev, pinning the patch keeps the deployed verifier from silently changing on a 1.26.x release, matching ADR-0035.

File: site-obsigna/public/verify/verify.js [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+      state = "Not anchored"; cls = "warn";
+    }
+    return (
+      '<div class="ring">' +
+        "<h3>External-anchor trust</h3>" +
+        '<div class="ring-state ' + cls + '">' + esc(state) + "</div>" +
+        "<p>" + esc(a.note || "") + "</p>" +
+      "</div>"
+    );
+  }
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply site-obsigna/public/verify/verify.js:196]: Fixed in 702eaa9. `anchorRing` now special-cases `result.verdict === "error"` and renders "Not evaluated" (mirroring `cryptoRing`), so an error response no longer shows a misleading "Not anchored" — consistent with the `note` ("not evaluated") that the WASM core already returns on error.

File: .github/workflows/cross-sdk-tests.yml [hunk 4]
```diff
@@ -7,12 +7,16 @@ on:
...
+      # The CLI<->WASM conformance gate compiles the web-verifier module, so a
+      # change there must re-run the gate.
+      - "web-verifier/**"
   pull_request:
     branches: [main]
     paths:
       - "cross-sdk-tests/**"
       - "sdk/go/**"
       - "spec/**"
+      - "web-verifier/**"
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply .github/workflows/cross-sdk-tests.yml:19]: Acknowledged. The workflow edits in this PR were made with the maintainer's explicit approval during this session, satisfying the review requirement. This file's change is path-filters only — adding `web-verifier/**` so the CLI↔WASM conformance gate re-runs whenever the verifier source changes.
